addons/web/static/tests/core/commands/command_service.test.js

```diff
@@ -0,0 +1,1214 @@
+        setup() {
+            useCommand("Take the throne", () => {
+                expect.step("Hodor");
+            });
+        }
+    }
+    const componentInfo = reactive({ Component: MyComponent });
+    await mountWithCleanup(Parent, { props: { componentInfo } });
+
+    // hoot-event: press doesn't work as expected for this
+    press("Control+k");
+    await animationFrame();
+    expect(".o_command").toHaveCount(1);
+    expect(".o_command").toHaveText("Take the throne");
+
+    await contains(".o_command").click();
+    expect(["Hodor"]).toVerifySteps();
+
+    componentInfo.Component = null;
+    await animationFrame();
+
+    press("Control+k");
+    await animationFrame();
+    expect(".o_command").toHaveCount(0);
+});
+
+test("useCommand hook when the activeElement change", async () => {
+    class MyComponent extends TestComponent {
+        setup() {
+            useCommand("Take the throne", () => {});
+            useCommand("Lose the throne", () => {}, { global: true });
+        }
+    }
+
+    class OtherComponent extends Component {
+        static template = xml`<div t-ref="active"><div tabindex="1">visible</div></div>`;
+        static props = ["*"];
+        setup() {
+            useActiveElement("active");
+            useCommand("I'm taking the throne", () => {});
+        }
+    }
+
+    await mountWithCleanup(MyComponent);
+    press("Control+k");
+    await animationFrame();
+    expect(".o_command").toHaveCount(2);
+    expect([...getFixture().querySelectorAll(".o_command")].map((e) => e.textContent)).toEqual([
+        "Take the throne",
+        "Lose the throne",
+    ]);
+    press("escape");
+    await animationFrame();
+
+    await mountWithCleanup(OtherComponent, { noMainContainer: true });
+    press("Control+k");
+    await animationFrame();
+    expect(".o_command").toHaveCount(2);
+    expect([...getFixture().querySelectorAll(".o_command")].map((e) => e.textContent)).toEqual([
+        "Lose the throne",
+        "I'm taking the throne",
+    ]);
+});
+
+test("useCommand hook with isAvailable", async () => {
+    let available = false;
+    class MyComponent extends TestComponent {
+        setup() {
+            useCommand("Take the throne", () => {}, {
+                isAvailable: () => {
+                    return available;
+                },
+            });
+        }
+    }
+    await mountWithCleanup(MyComponent);
+
+    press("Control+k");
+    await animationFrame();
+    expect(".o_command_palette").toHaveCount(1);
+    expect(".o_command").toHaveCount(0);
+
+    press("escape");
+    await animationFrame();
+    available = true;
+    press("Control+k");
+    await animationFrame();
+    expect(".o_command_palette").toHaveCount(1);
+    expect(".o_command").toHaveCount(1);
+});
+
+test("command with hotkey", async () => {
+    const hotkey = "a";
+    env.services.command.add("test", () => expect.step(hotkey), {
+        hotkey,
+    });
+    await animationFrame();
+
+    press("a");
```

Comment:
to activate the hotkey, one must press 'a' or 'shift-a'? I though 'shift-a', but now I'm not sure. I hope the press helper doesn't add 'shift' automatically.

Reply:
in this test, there is the pure command A with hotkey "a", and B and C that are hotkeys in the DOM, automatically registered as "alt+[bc]".
press does work as expected.

The original test used triggerHotkey, which, when the second parameter was true, added the alt key stroke artificially